File: src/main/java/com/example/eventplanner/model/Poll.java
package com.example.eventplanner.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.List;
import java.util.Set;

@Entity
@Getter
@Setter
public class Poll {

    public enum PollType {
        SINGLE_CHOICE,
        MULTIPLE_CHOICE
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    private Event event;

    @Column(nullable = false)
    private String question;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private PollType pollType;

    @ElementCollection
    private List<String> options;

    @ElementCollection
    private List<Integer> votes;

    @ManyToMany(fetch = FetchType.LAZY,
            cascade = {
                    CascadeType.PERSIST,
                    CascadeType.MERGE
            },mappedBy = "votedPolls")
    private Set<User> voters;

}

File: frontend/src/app/entities/poll.ts
import {Event} from "./event";
import {User} from "./user";

export enum PollType {
  SINGLE_CHOICE = 'SINGLE_CHOICE',
  MULTIPLE_CHOICE = 'MULTIPLE_CHOICE'
}

export interface Poll {
  id?: number;
  event?: Event;
  question: string;
  pollType: PollType;
  options: string[];
  votes: number[];
  voters: User[];
}
